File: Linked List/linkedlist.c
#include "stdlib.h"
#include "stdio.h"

struct linkedList
{
    int data;
    struct linkedList *next;
};

void insert(struct linkedList **list, int value)
{
    struct linkedList *node = malloc(sizeof(value));

    node->data = value;
    node->next = *list;
    *list = node;
}

void delete(struct linkedList **list, int value)
{
    struct linkedList *items = *list;
    if (*list == NULL)
    {
        printf("error, List is empty\n");
        return;
    }

    else if (items->data == value)
    {

        if (items->next != NULL)
        {

            items->data = items->next->data;
            items->next = items->next->next;
        }
        else
        {
            free(items);
            free(list);
        }
    }

    else
    {
        struct linkedList *current = items->next;
        struct linkedList *prev = items;

        while (current != NULL)
        {

            if (current->data == value)
            {

                struct linkedList *temp = current;
                prev->next = current->next;
                free(temp);
                break;
            }
            prev = current;
            current = current->next;
        }
    }
}

struct linkedList *traverse(struct linkedList **list)
{

    if (list == NULL)
    {
        printf("Error, List is empty\n");
        return NULL;
    }
    struct linkedList *current = *list;

    while (current != NULL)
    {

        printf("%d\n", current->data);

        current = current->next;
    }
}

void freeList(struct linkedList **list)
{

    struct linkedList *items = *list;
    struct linkedList *next;

    while (items != NULL)
    {

        next = items->next;
        free(items);
        items = next;
    }

    free(list);
}

int main(int argc, char **argv)
{

    struct linkedList *list = NULL;

    insert(&list, 1);
    insert(&list, 2);
    insert(&list, 3);
    insert(&list, 4);

    traverse(&list);

    delete (&list, 2);
    printf("\n");

    traverse(&list);

    freeList(&list);

    traverse(&list);
}

File: Linked List/linkedlist.py
import copy


class LinkedList:

    def __init__(self, value=None, next=None):

        self.value = value
        self.next = next

    def addNode(self, value):

        newNode = LinkedList(value)

        temp = copy.deepcopy(self)
        self.value = value
        self.next = temp

    def delete(self, value):
        if self.value == None:
            print("Error list is empty")
            return

        if (self.next != None and self.value == value):

            self.value = self.next.value
            self.next = self.next.next
            return

        temp = self
        prev = self
        while (temp != None and temp.value != value):

            prev = temp
            temp = temp.next

        if (temp == None):
            print("Error, value not found")
            return

        prev.next = temp.next

    def traverse(self):

        current = self
        if (current.value == None):
            print("Error, list is empty")

        while current.next != None:

            print(current.value)
            current = current.next


linked = LinkedList()

linked.addNode(1)


linked.traverse()

linked.delete(1)
print("\n")

linked.traverse()
